### PR TITLE
BGP Parameters

### PR DESCRIPTION
**Pull request by sbarguil**

### CODE REVIEW

**Comment by boucadair** [comment]
Add more motivation to explain why this is not left at the device level and why we need to manipulate it at the network model.

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 1]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+                           "Define whether AS override is enabled, i.e., replace
+                           the AS number of the peer specified in the AS Path
+                           attribute with the local AS number.";
+                        }
+
+                      leaf default-route {
+                        type boolean;
+                        default false;
+                        description
+                         "Define whether default route can be advertised
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1422]: "Define whether default route can be advertised to its peer."; --> 
"If enabled, the default route is advertised to its peer.

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 2]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+                         to its peer.";
+                        }
+
+
+                      container bgp-max-prefix {
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1427]: Add description. For example
"Maximimum number of BGP prefixes allowed in the BGP session. Allows to control how many prefixes can be received from a neighbor. If the limit is exceeded, the session will be teared down."
Why 5000? Any reference to support it?
Any reference to the parameter? RFC4271 ?

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 3]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+                        leaf max-prefix {
+                          type uint32;
+                          default 5000;
+                        }
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1431]: what is the difference between max-prefix and bgp-max-prefix?

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 4]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+
+                      leaf warning-threshold{
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1433]: Add description and reference
"Enables warining when number of received messages exceeds the warning-threshold"

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 5]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+                        type decimal64 {
+                          fraction-digits 5;
+                          range "0..100";
+                        }
+                          units percent;
+                        default 75;
+                        }
+
+                      leaf violate-action {
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1442]: add reference  warning-threshold 
Add comment that the behaviour is to be passed to the BGP configuration

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 6]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+                           "bpg neighbour max-prefix violate action";
+                       }
+
+                      leaf restart-interval {
+                        type uint16;
+                          description "Time interval (min) after which peering session will be reestablished";
+                          }
+                        }
+
+                      container bgp-timer {
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1464]: Add reference and description for the timers. See  draft-ietf-idr-bgp-model-09.
   "RFC 4271, Section 4.4.?
            RFC 4271, Section 10."?;

File: yang/IETF-04/ietf-l3vpn-ntw.yang [hunk 7]
```diff
@@ -1406,6 +1406,74 @@ module ietf-l3vpn-ntw {
...
+                            units "seconds";
+                        }
+
+                      leaf hold-time {
+                          type uint16;
+                          default 180;
+                          units "seconds";
+                        }
+                      }
                       description
```
Comment (oscargdd) [inline yang/IETF-04/ietf-l3vpn-ntw.yang:1477]: Add comment that this does not cover all possible BGP configurations, but just the more relevant ones.

**Pull request merged**